dataloader.go

```diff
@@ -179,7 +179,11 @@ func (l *Loader) Load(key string) Thunk {
 	}
 
 	thunk := func() (interface{}, error) {
-		if result.value == nil {
+		result.mu.RLock()
```

Comment:
I think you're correct. From my understanding of Go concurrency, 

- the acquisition of the single write lock will prevent two goroutines from reading the only message in the channel. The second goroutine will return `false` if the channel is closed and has no results left. 
- that is (almost) guaranteed to happen because the second goroutine will block on `v, ok := <-c` until it receives a second result (shouldn't happen, but if it somehow does, the second result will be used) or the channel is closed (will happen, or a test which waits for both goroutines to complete will fail due to a deadlock.)

FWIW, my original concern was "if I can make a test that fails with -race enabled, then my code, which depends on the concurrent usage of thunks, will not be able to run tests with -race". Maybe there's a cleaner way to solve my problem?